Bug Fix - stop crashing when copying a tar with an ImageIndex

DennisDenuto opened the pull request:
Opening as a draft as there is still work to do. Namely

- [x] backfill test copying a tarfile with an ImageIndex

cppforlife commented on pkg/imgpkg/cmd/copy.go:
```diff
@@ -242,6 +242,10 @@ func (o *CopyOptions) writeBundleLockOutput(bundle *bundle.Bundle) error {
 func processedImagesMediaType(processedImages *ctlimgset.ProcessedImages) []string {
 	everyMediaType := []string{}
 	for _, image := range processedImages.All() {
+		if image.Image == nil {
```
dont we need to recurse into image index as well?

DennisDenuto replied:
We do! we discovered this while TDD'ing. See updated impl.

cppforlife commented on pkg/imgpkg/cmd/copy.go:
```diff
@@ -242,17 +243,42 @@ func (o *CopyOptions) writeBundleLockOutput(bundle *bundle.Bundle) error {
 			}
-			everyMediaType = append(everyMediaType, string(mediaType))
+			for _, descriptor := range indexManifest.Manifests {
+				image, err := image.ImageIndex.Image(descriptor.Digest)
```
i might be wrong, i thought that indexes could contain other indexes as well.

DennisDenuto replied:
Added logic to recurse through image indexes

DennisDenuto committed:
Bug Fix

- This prevent a nilpointer error from occuring during a copy
- When copying a tarfile with an imageindex, stop assuming it is always an image (it might be an imageindex)
- Backfill e2e tests

Authored-by: Dennis Leon <leonde@vmware.com>

joaopapereira reviewed (approved):
LGTM
